### PR TITLE
Sync HAFS submodules as of 01/26/2022

### LINKED ISSUE #122: Sync HAFS submodules as of 202201
## Description
Provide a clear and concise description of the requested feature/capability.
Sync HAFS submodules (hafs_forecast.fd, hafs_post.fd, hafs_utils.fd) with their latest authoritative branches as of 202201.

## Proposed solution
How should the new feature/capability be added? If you have thoughts on the implementation strategy, please share them here.

- Sync HAFS submodules hafs_forecast.fd and hafs_utils.fd with their latest authoritative branches as of 202201.
- Symbolically link sorc/hafs_forecast.fd/FV3/upp into sorc/hafs_post.fd so that hafs_post.fd points to the same commit or UPP develop branch.

## Related to
Directly reference any related issues or PRs in this or other repositories, and describe how they are related. Examples:
- fixed by hafs-community/hafs/pull/<pr_number>

### PR DESCRIPTION
**Pull request by BinLiu-NOAA**
## Description of changes
Sync HAFS submodules (hafs_forecast.fd, hafs_post.fd, hafs_utils.fd) with their latest authoritative branches as of 01/26/2022.
 - Update submodule hafs_forecast.fd to point to the ufs-weather-model develop branch as of 01/26/2022.
 - Remove hafs_post.fd (UPP) as a direct submodule of HAFS, and use the same one used by ufs-weather-model/FV3atm 
 (hafs_forecast.fd/FV3/upp).
 - Sync hafs_utils.fd with its latest develop branch (as of 01/26/2022)
 - Rename sorc/hafs_vortextracker.fd into sorc/hafs_tracker.fd.
 - Remove sorc/hafs_vortexinit.fd/.gitkeep.

## Issues addressed (optional)
If this PR addresses one or more issues, please provide link(s) to the issue(s) here.
- fixes hafs-community/HAFS/issues/122

## Tests conducted

So far build on Orion succeeded. Have not run any regression tests yet.

## Application-level regression test status
Running the HAFS application-level regression tests is currently performed by code reviewers after the developer creates the initial PR. As regression tests are conducted, the testers should use the checklist below to indicate **successful** regression tests. You may add other tests as needed. If a test fails, do not check the box. Instead, describe the failure in the PR comments, noting the platform where the test failed.

- [x] Jet
- [x] Hera
- [x] Orion
- [x] WCOSS Dell P3
- [x] WCOSS Cray

Again, appreciate your help to conduct the HAFS application/workflow level regression tests for this PR.
@evankalina, @panll, @mrinalbiswas (Hera and Orion)
@JunghoonShin-NOAA (Jet)
@LinZhu-NOAA (wcoss_dell_p3)
@BijuThomas-NOAA (wcoss_cray)

### CODE REVIEW

**Comment by BijuThomas-NOAA** [comment]
@BinLiu-NOAA 
Regression tests (feature/hafs_sync_202201 branch) were successful on WCOSS-Cray.

**Review by BijuThomas-NOAA — APPROVED** [review]

**Comment by panll** [comment]
Regression tests passed successfully on Hera. @BinLiu-NOAA @evankalina

**Review by panll — APPROVED** [review]

**Comment by JunghoonShin-NOAA** [comment]
Regression tests on Jet completed successfully on Jet.

**Review by JunghoonShin-NOAA — APPROVED** [review]

**Review by mrinalbiswas — APPROVED** [review]
@BinLiu-NOAA RTs passed on Orion.

**Comment by LinZhu-NOAA** [comment]
@BinLiu-NOAA Regression tests completed successfully on WCOSS-DELL.

**Review by LinZhu-NOAA — APPROVED** [review]

**Comment by BinLiu-NOAA** [comment]
Thanks a lot for the quick response and finishing the RTs on all HAFS supported platforms! Much appreciated! Waiting for another approval from @evankalina, @KathrynNewman or @ZhanZhang-NOAA. Then it is ready to merge.

**Review by evankalina — APPROVED** [review]

**Pull request merged**
